src/lib/Microsoft.Fx.Portability/Utils/JsonConverters/JsonToStringConverter.cs

```diff
@@ -4,23 +4,26 @@
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using System;
+using System.Collections.Concurrent;
 
 namespace Microsoft.Fx.Portability.Utils.JsonConverters
 {
     internal class JsonToStringConverter<T> : JsonConverter<T>
     {
+        private readonly ConcurrentDictionary<string, T> _cache;
         private readonly Func<string, T> _factory;
 
         public JsonToStringConverter(Func<string, T> factory)
         {
+            _cache = new ConcurrentDictionary<string, T>(StringComparer.Ordinal);
             _factory = factory;
         }
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
             var value = serializer.Deserialize<JToken>(reader).ToString();
 
-            return string.IsNullOrEmpty(value) ? default(T) : _factory(value);
+            return string.IsNullOrEmpty(value) ? default : _cache.GetOrAdd(value, _factory);
```

Comment:
we still want "default(T)" instead of "default", right?

Reply:
`default` is the literal keyword in C# 7.1 for `default(T)`
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-1#default-literal-expressions